Core/include/Acts/Utilities/AngleHelpers.hpp

```diff
@@ -9,16 +9,51 @@
 #pragma once
 
 #include <cmath>
+#include <numbers>
 
 namespace Acts::AngleHelpers {
 
+template <typename Scalar>
+struct EtaThetaConversionTraits {
+  static constexpr Scalar minTheta = 0;
+  static constexpr Scalar maxTheta = std::numbers::pi_v<Scalar>;
+
+  static constexpr Scalar minEta = -std::numeric_limits<Scalar>::infinity();
+  static constexpr Scalar maxEta = std::numeric_limits<Scalar>::infinity();
+};
+
+template <>
+struct EtaThetaConversionTraits<float> {
+  static constexpr float minTheta = 0;
+  static constexpr float maxTheta = std::numbers::pi_v<float> - 1e-6f;
```

Comment:
Why are these all specialised? I guess this is still experimental...

How about this:
```
template <Float>
struct EtaThetaConversionTraits<Float> {
  static constexpr Float minTheta = std::numeric_limits<Float>::min();
  static constexpr Float maxTheta = std::numbers::pi_v<Float> - std::numeric_limits<Float>::epsilon();
  static constexpr Float minEta = -std::numeric_limits<Float>::infinity();
  static constexpr Float maxEta = std::numeric_limits<Float>::infinity();
};
```

I checked `maxTheta<float>` gives &eta;=-16.39916 and `maxTheta<double>` gives &eta;=-37.33185619326892, which look reasonable (as does `long double`). I don't think it is possible to find a value of `maxTheta` that gives &eta;=`-inf`, but at least we can ensure it is not `nan`.

For `minTheta`, I think `0` (&eta;=`inf`) or `std::numeric_limits<Float>::min()` (&eta;=88.02969) are both reasonable.

Note that `epsilon()` is only strictly correct at 1.0, but it seems to do something reasonable for &pi;.

You could also do something clever with `concept`s for better error messages if ever used for non-FP types, if you want.

Reply:
I was experimenting with it and quite surprised that almost all the values worked out of the box. Usually `float` and `double` behave a little different so I am not sure if the `epsilon` will work. The idea was to capture the "last" value which works before the operations would throw FPEs. For `float` I found this boundary while for all others it worked out of the box.

Haven't checked if it throws FPEs yet.

Reply:
I have been thinking about this... One problem is that these details of the FP rounding at the limits of precision are quite fragile. They could change with compiler or library version, or CPU type (x86_64, ARM, or even GPU) or version. I was testing with gcc godbolt on x86_64. Maybe you were using clang on macOS ARM?

Whatever. I don't know much about forward physics, but I don't think people would be using pseudorapidity in the very forward region. If we limit &eta; anywhere &NestedGreaterGreater;4, we should be fine. Can I propose something like
```
static constexpr Float thetaFromEtaNoChecked(Float eta) { return 2 * std::atan(std::exp(-eta)); }
static constexpr Float maxEta = 10;  // arbitrary large eta
static constexpr Float minEta = -maxEta;
static constexpr Float minTheta = thetaFromEtaNoChecked(maxEta);
static constexpr Float maxTheta = thetaFromEtaNoChecked(minEta);
```
and then include the checks in `thetaFromEta()` and `etaFromTheta()` substituting the `minTheta` etc if outside the limits.

One problem with this simple definition is that `std::atan` and `std::exp` aren't defined `constexpr` until (probably) C++26. gcc doesn't complain, but clang does. For the moment, I think it's fine to put in
```
static constexpr Float minTheta = static_cast<Float>(9.0799859462585550026e-05L);  // 2*std::atan(std::exp(maxEta))`
static constexpr Float maxTheta = static_cast<Float>(3.1415018537303306529L);  // 2*std::atan(std::exp(minEta))`
```
This won't be exact (what is?), but since they are limits, I don't think it really needs to be.

Reply:
I think the round trip will not produce the same numbers anyways so we do not need to be too precise with the boundaries. I thought it would be nice to extend the range based on the floating point type since it might actually reach further with double but maybe you are right and there is no use case in converting these shallow angles anyways and one can just cut at 10.